diff --git a/CMakeLists.txt b/CMakeLists.txt
index 38e069e..4eb28de 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -260,9 +260,9 @@ else()
 	add_library(Usrsctp::Usrsctp ALIAS usrsctp)
 endif()
 
-configure_file (
-    ${PROJECT_SOURCE_DIR}/cmake/version.h.in
-	${CMAKE_CURRENT_SOURCE_DIR}/include/rtc/version.h
+file(COPY
+    ${CMAKE_CURRENT_SOURCE_DIR}/cmake/version.h.in
+	DESTINATION ${CMAKE_CURRENT_SOURCE_DIR}/include/rtc/version.h
 )
 
 add_library(datachannel SHARED
