--- a/src/google/protobuf/parse_context.h
+++ b/src/google/protobuf/parse_context.h
@@ -966,14 +966,14 @@
   return {info.p, result};
 }
 
-static const char* VarintParseSlowArm(const char* p, uint32_t* out,
+inline const char* VarintParseSlowArm(const char* p, uint32_t* out,
                                       uint64_t first8) {
   auto tmp = VarintParseSlowArm32(p, first8);
   *out = tmp.second;
   return tmp.first;
 }
 
-static const char* VarintParseSlowArm(const char* p, uint64_t* out,
+inline const char* VarintParseSlowArm(const char* p, uint64_t* out,
                                       uint64_t first8) {
   auto tmp = VarintParseSlowArm64(p, first8);
   *out = tmp.second;
